agenda/data/src/main/java/com/gkp/agenda/data/notification/AgendaItemReminderNotification.kt

```diff
@@ -0,0 +1,81 @@
+package com.gkp.agenda.data.notification
+
+import android.Manifest
+import android.app.NotificationChannel
+import android.app.NotificationManager
+import android.app.PendingIntent
+import android.app.TaskStackBuilder
+import android.content.ComponentName
+import android.content.Context
+import android.content.Intent
+import android.content.pm.PackageManager
+import android.os.Build
+import androidx.annotation.RequiresApi
+import androidx.core.app.ActivityCompat
+import androidx.core.app.NotificationCompat
+import androidx.core.app.NotificationManagerCompat
+import androidx.core.net.toUri
+import com.gkp.agenda.data.R
+import com.gkp.agenda.domain.model.AgendaItemType
+import com.gkp.agenda.domain.notification.DEEP_LINK_URI
+import com.gkp.agenda.domain.notification.NotificationInfo
+import com.gkp.agenda.domain.notification.ReminderNotification
+
+const val REMINDER_CHANNEL_ID = "REMINDER_CHANNEL"
+const val MAIN_ACTIVITY_COMPONENT = "com.gkp.tasky.MainActivity"
+
+@RequiresApi(Build.VERSION_CODES.O)
+class AgendaItemReminderNotification (
+    private val context: Context
+) : ReminderNotification {
+
+    private val notificationManager = NotificationManagerCompat.from(context)
+    init {
+        val notificationChannel = NotificationChannel(
+            REMINDER_CHANNEL_ID,
+            "Reminder",
+            NotificationManager.IMPORTANCE_DEFAULT
+        )
+        notificationManager.createNotificationChannel(
+            notificationChannel
+        )
+    }
+
+    override fun showNotification(notificationInfo: NotificationInfo) {
+
+        val intent = Intent().apply {
+            component = ComponentName(
+                context.packageName,
+                MAIN_ACTIVITY_COMPONENT
```

Comment:
Instead of relying on a string value which won’t work anymore when you rename or relocate your Activity class, rather use reflection

```
Intent(context, MainActivity::class.java).apply {
  this.data = "tasky://agenda_detail/${notificationInfo.id}/${AgendaItemType.REMINDER}".toUri()
  addFlags(Intent.FLAG_ACTIVITY_SINGLE_TOP)
}
```